### PR TITLE
CB-7395 Revise, restrict and POC Azure minimal permission list

### PR DESCRIPTION
**Pull request by gergopapi2**
Permission set for single RG use-case not fully tested => draft.

Question: given the most likely scenario from a customer is to bring its own RG, network and storage account, we could create a typical set of permissions that does not contain creation and deletion of such entities. What do you think?

### CODE REVIEW

File: cloud-azure/src/main/resources/definitions/azure-role-def-single-rg.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  "NotActions": [],
+  "DataActions": [
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/delete",
+    "Microsoft.Storage/storageAccounts/blobServices/containers/blobs/add/action"
+  ],
+  "NotDataActions": [],
+  "AssignableScopes": [
+    "/subscriptions/{subscriptionId}"
```
Comment (pdarvasi) [inline cloud-azure/src/main/resources/definitions/azure-role-def-single-rg.json:82]: We should scope this on RG level, shouldn't we?

Comment (gergopapi2) [reply cloud-azure/src/main/resources/definitions/azure-role-def-single-rg.json:82]: Yes, definitely, thank you!

Comment (gergopapi2) [reply cloud-azure/src/main/resources/definitions/azure-role-def-single-rg.json:82]: Hm, but actually I am not sure if we should decrease assignableScope. What if the customer wants to use the same role definition for multiple resource groups? I think the assignableScope can be as wide as possible, whereas the real assignment should be as tight as possible.

Comment (pdarvasi) [reply cloud-azure/src/main/resources/definitions/azure-role-def-single-rg.json:82]: The customers who are asking for single RG feature will most probably scope their credential to have access on RG level, and the AssignableScopes can contain comma-separated list of RG-s

**Pull request merged**